ROADMAP.md

```diff
@@ -1,62 +1,191 @@
+
+## 🛣️ Milestones
+
+### 2022
+
+#### Early Q4 (October)
+- [A.1 User Configured Interop Tests & Dashboard](#1-user-configured-interop-tests--dashboard)
+
+#### Mid Q4 (November)
+- [A.2 Interop tests for all existing/developing features](#2-interop-tests-for-all-existingdeveloping-features)
+- [A.3 Rock solid libp2p releases](#3-rock-solid-libp2p-releases)
+- [C.1 Benchmarking using nix-builders](#1-benchmarking-using-nix-builders)
+
+#### End of Q4 (December)
+- [A.4 Canonical Interop Tests & Dashboard](#4-canonical-interop-tests--dashboard)
+
+### 2023
+
+#### Early Q1 (January)
+
+- add here
+
+#### Mid Q1 (February)
+
+- add here
+
+#### End of Q1 (March)
+- [C.2 Benchmarking using remote runners](#2-benchmarking-using-remote-runners)
+
+### Up Next
+
+## 📖 Appendix
+
+**Projects are listed in descending priority.**
+
+### A. [Multi-dimensional Testing/Interop Visibility](https://github.com/libp2p/test-plans/issues/53)
+**Why:** libp2p supports a variety of protocols (multiple transports, muxers, & security protocols) across implementations in different languages. Until we actually test them together, we can't guarantee implementation interoperability. We need to ensure and demonstrate that: interoperable features work with each other as expected and we don't introduce degradations that break interoperability in new releases.
+
+**Goal:** libp2p implementers run tests across permutations of libp2p implementations, versions, and supported protocols. Implementers and users can reference a single website with a dashboard to view the Pass/Fail/Implemented/Not Implemented results of multi-dimensional tests.
+
+**How:**
+#### 1. [User configured interop tests & dashboard](https://github.com/libp2p/test-plans/issues/55)
+Enable test-plans maintainers to define a configuration (of libp2p impls, versions, protocols, expected RTT), run Testground tests per configuration, and retrieve resulting data in a standard format.
+The test case results can be formatted as a "dashboard" (simple Markdown table of Pass/Fail results.)
+
+#### 2. Interop test plans for all existing/developing features
+<!-- TODO: Create issue -->
+Using tooling from A.1, all features of go-libp2p and rust-libp2p that should be interoperable are tested (i.e. transports (TCP, QUIC, WebRTC, WebTransport), multiplexers (mplex, yamux), secure channels (TLS, Noise), etc.) across versions.
+
+Features currently in development across implementations (like WebRTC in go-libp2p and rust-libp2, or QUIC & TLS in rust-libp2p) are not merged without at least manually running these test suites.
+
+Test suites are run in libp2p/test-plans CI.
+
+**Note:** node.js-libp2p is not listed here since it does not have support in Testground yet.
+
+#### 3. Rock solid libp2p releases
+<!-- TODO: Create issue -->
+The full `libp2p/test-plans` test suite is run before releasing a version of go-libp2p,  rust-libp2p, and js-libp2p.
```

Comment:
I think run the interoperability testplan should occur today (or at least soon) before releases.  It's fine if it's manual. 

Can we pull this portion in sooner?

Reply:
> I think run the interoperability testplan should occur today (or at least soon) before releases.

I'm having trouble parsing this sentence.

> It's fine if it's manual.

The problem is that we have many parts of our stack that are currently untested (other than unit tests): hole punching, autonat, autorelay etc. We manually test these parts when we make changes to the respective code paths, but we don't rerun these manual tests on every release. It's just too much stuff to test.
This is not ideal, and we should have automated interoperability tests for all these components. Requiring manual testing though would add hours of testing effort to our release process.

Reply:
I think we all agree that these tests are needed like yesterday. But we don’t have this yet, so what’s the fastest way we can get to the world where we have this? I’m afraid manual smoke testing will cut into the effort to build this. 

I’m also afraid that adding more friction to the release process will cause patches to be unnecessarily delayed. 

If we weren’t prioritizing these projects, I would agree that we should _at least_ have a manual test checklist. But since we are, it seems to me we can live with the status quo for a bit longer if it gets us this faster.

Reply:
@marten-seemann @MarcoPolo : My bad on incoherent sentence.  Fixed now.

Correct me if this is wrong, but my understanding is that we have some degree of interoperability testing run as part of CI but there is also a "big test" that doesn't run as part of CI because it takes too long.

What I'm suggesting we do today in our release process is run the big test manually as one of the [release checklist items](https://github.com/libp2p/go-libp2p/blob/master/.github/ISSUE_TEMPLATE/release.md).  

I'm not suggesting we do any manual smoke testing features that don't have a test-plan (unless of course we made changes in one of those areas).  I'm suggesting we do make sure we execute the test-plan we have written that is flexing interoperability.  

In future, I expect the "big interoperability test" either gets run as part of CI (because we have a cluster than can execute it quickly enough) or it runs as a nightly cron job and the release process is one of verifying that last night's run passed.

Does that make sense?

Reply:
Removed "3. Rock solid libp2p releases" and combined with "2. Interop test plans for all existing/developing libp2p transports"

3's intention was to run the transport interop tests as CI nightly job, serving as regression tests for new release candidates.